src/DetailsView/components/load-assessment-button.tsx

```diff
@@ -3,31 +3,125 @@
+        if (this.state !== undefined && this.state.loadedAssessmentData != null) {
+            return {
+                id: this.state.loadedAssessmentData.assessmentData.persistedTabInfo.id,
+                url: this.state.loadedAssessmentData.assessmentData.persistedTabInfo.url,
+                title: this.state.loadedAssessmentData.assessmentData.persistedTabInfo.title,
+            };
+        }
+
+        return {
+            id: this.props.tabStoreData.id,
+            url: this.props.tabStoreData.url,
+            title: this.props.tabStoreData.title,
+        };
+    }
+
+    public getAssessmentForLoad() {
+        const input = this.props.deps.document.createElement('input');
+        input.type = 'file';
+        input.accept = '.a11ywebassessment';
+
+        const onReaderLoad = (readerEvent: ProgressEvent<FileReader>) => {
+            const content = readerEvent.target.result as string;
+            this.setState(_ => ({
+                loadedAssessmentData: this.props.deps.assessmentDataParser.parseAssessmentData(
+                    content,
+                ),
+            }));
+            if (this.state.prevTab != null) {
+                this.toggleLoadDialog();
+            } else {
+                this.props.deps.detailsViewActionMessageCreator.loadAssessment(
+                    this.state.loadedAssessmentData,
+                    this.state.newTab.id,
+                );
+            }
+        };
+        const onInputChange = (e: Event) => {
+            const file = (e.target as HTMLInputElement).files[0];
+            this.props.deps.fileReader.onload = onReaderLoad;
+            this.props.deps.fileReader.readAsText(file, 'UTF-8');
+        };
+
+        input.onchange = onInputChange;
+        input.click();
+    }
+
     public render(): JSX.Element {
         return (
-            <InsightsCommandButton
-                data-automation-id={loadAssessmentButtonAutomationId}
-                iconProps={{ iconName: 'FabricOpenFolderHorizontal' }}
-                onClick={() => this.props.deps.loadAssessmentHelper.getAssessmentForLoad()}
-            >
-                Load assessment
-            </InsightsCommandButton>
+            <>
+                <InsightsCommandButton
+                    data-automation-id={loadAssessmentButtonAutomationId}
+                    iconProps={{ iconName: 'FabricOpenFolderHorizontal' }}
+                    onClick={() => this.getAssessmentForLoad()}
+                >
+                    Load assessment
+                </InsightsCommandButton>
+
+                <LoadAssessmentDialog
+                    {...this.props}
+                    tabId={this.props.tabStoreData.id}
+                    loadedAssessmentData={this.state.loadedAssessmentData}
+                    prevTab={this.state.prevTab}
+                    newTab={this.state.newTab}
+                    show={this.state.show}
+                    onClose={this.toggleLoadDialog.bind(this)}
```

Comment:
FYI: this is not necessary. We can just make toggleLoadDialog an arrow method and pass in appropriately.